canonical-web-and-design/maas-ui
# feat: update locations for LXD host/cluster actions

## Done

- Updated locations of refresh and compose form buttons for LXD single hosts and clusters
- Converted a couple of component test files to RTL

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the details page of a LXD single host
  - Check that clicking "Refresh host" in the header opens the refresh host form
  - Check that clicking "Add VM" in the table opens the compose form
  - Check that clicking "Delete VM" after selecting some machines opens the delete machine form
- Go to the details page of a LXD cluster
  - Check that clicking "Refresh cluster" in the header opens the refresh cluster form
  - Go to the Virtual Machines tab and check there is no button to "Add VM"
  - Go to the KVM hosts tab and click on a host
  - Check that clicking "Add VM" opens the compose form for that host

## Fixes

Fixes #3948 

## Screenshots

### LXD single host details
![Screenshot 2022-06-13 at 12-19-33 bolla virtual machines bolla MAAS](https://user-images.githubusercontent.com/25733845/173267998-0dd288cc-7a80-49de-8702-ef1bcc481240.png)


### LXD cluster details
![Screenshot 2022-06-13 at 12-19-46 karura-polong-20211104 KVM hosts bolla MAAS](https://user-images.githubusercontent.com/25733845/173268016-bc2d9712-7f0d-47c3-a30e-72662373e174.png)


### LXD cluster host details
![Screenshot 2022-06-13 at 12-19-58 polong in karura-polong-20211104 virtual machines bolla MAAS](https://user-images.githubusercontent.com/25733845/173268048-ca66fd64-dae9-4c3c-8791-904f38cbed2c.png)
